### ISSUE TITLE
Can't login after upgrade matomo 3.12.0

### ISSUE THREAD

**Issue by metfan** [matomo-org/matomo#15083]
I can't login anymore after upgrading matomo from 3.9 to 3.12. 
I did db upgrade and get stuck outside the app.
Tested with Firefox 70 and Brave 0.70 based on chromium 78

Here error message I get
![image](https://user-images.githubusercontent.com/1121867/67897057-eb511280-fb5d-11e9-8521-8512c2b7d68e.png)

I tried to reset my password following this page https://matomo.org/faq/how-to/faq_191/ but I still get the same error.

I'm using same computer, same browser than yesterday, no proxy....

thanks for your help.

**Comment by metfan** [comment]
hum, finally I got it work on Microsoft Edge but I still get the error on Firefox and Brave.

**Comment by metfan** [comment]
Hello @JugurthaK , 
I don't have adblocker on Brave, I disable brave shield but I still have the problem.
I tried on Firefox after I disable uBlock Origin and built in Firefox shield and still have the problem.

Strange it work on Microsoft Edge.

Thanks for your help.

**Comment by xshadow** [comment]
Same issue here. My Firefox 70.0.1 (64-bit), sends  `Origin: null`to Matomo, which results in this error. That also happens to me on a plain FF installation. 

There are no problems with Safari or TorBrowser, they send `Origin: https://analytics.foobar.org` properly.

**Comment by tsteur** [comment]
AFAIK some users had issues updating to 3.10 or so. Maybe clearing all cookies helps? Does it work in incognito mode?

I wonder why not more users are experiencing this issue on Firefox or so. Any thoughts? @Findus23 @mattab ?

**Comment by xshadow** [comment]
Yeah it seems to be like an odd behavior. I would also suspect plugins. I also had this issue before upgrading to 3.12, so it does not has to be introduced in the current release. 

Deleting all cookies did not help. 
Incognito mode with plugins enabled does not work.

**Comment by xshadow** [comment]
I  can reproduce the problem. If you set in your Apache configuration the following header `        Header set Referrer-Policy "no-referrer"`then you are not able to login with a cache cleared FF (all plugins enabled),

**Comment by tsteur** [comment]
@xshadow would apache always overwrite that header? Or would apache keep a header that we set in Matomo (cause I think we do set a referrer-policy AFAIK)

**Comment by xshadow** [comment]
As far as I understand the docs "your" header will be [overwritten](https://httpd.apache.org/docs/current/mod/mod_headers.html): 

> set
> The response header is set, replacing any previous header with this name.

**Comment by mattab** [comment]
Is there anything we could do about this, maybe detect this problem and mention it in the red error message like `We detected your web server returned a HTTP header "Referrer-Policy" with the value "no-referrer", which is not compatible with Matomo. Please change your webserver configuration to not set this header "Referrer-Policy"`

**Comment by rahaalizade** [comment]
hello,
I have the same problem with nginx 
I also tried to set the mentioned header and even remove it
but it didn't work and still gave me this error:
` If you previously connected using HTTPS, please ensure you are connecting over a secure (SSL/TLS) connection and try again.`

**Comment by srinivasmummareddy** [comment]
@fatemealizade  Have you found any solution for this issue?

**Comment by pr0digy7** [comment]
I have this issue too. I see that it's considered like "not a bug but problem with user servers" so it won't be fix.. But why couldnt we set our Referrer-Policy equal to "no-referrer" if we want to ? (it improves security).. Why is Matomo the only self hosted tool that won't allow this setting ?

**Comment by sgiehl** [comment]
@FabThelead Is there any specific reason why you need to use `Referrer-Policy: no-referrer` for Matomo?
In Matomo we are actually using the Referrer header for security purpose. For critical actions we require the origin from the request to come from a trusted host. We could add an option to disable that behavior, but it would downgrade the security.

Also please note that Matomo itself already sets the no-referrer option automatically for all external links for improved privacy. So forcing it on Apache side should actually not have any additional benefit/impact.

**Comment by pr0digy7** [comment]
Hi @sgiehl , for this project, the Referrer-Policy: no-referrer is set up at the domain level, on the server, and Matomo is installed in a subfolder of the website.

"We could add an option to disable that behavior" It could be a nice addition. In our case, the website is managed by sys admins and parent company and these kind of security changes, at the server level, can be tricky to ask/get (internal policies, IT being rigid about changes, main company not wanting that anyone change security rules, etc...)

Maybe it would be also nice to update the error message because the issue was tricky to find and replicate..

**Comment by MatomoForumNotifications** [comment]
This issue has been mentioned on **Matomo forums**. There might be relevant details there:

https://forum.matomo.org/t/moved-installation-and-no-access/64616/10
